### PR TITLE
Excise credit #186927846

### PR DESCRIPTION
**Pull request by embarnard**

### CODE REVIEW

File: app/forms/state_file/az_excise_credit_form.rb [hunk 1]
```diff
@@ -1,13 +1,52 @@
...
-    validates :ssn_no_employment, inclusion: { in: %w[yes no], message: I18n.t("errors.messages.blank") }
-    validates :household_excise_credit_claimed, inclusion: { in: %w[yes no], message: I18n.t("errors.messages.blank") }
+    validates :ssn_no_employment, inclusion: { in: %w[yes no], message: :blank }
+    validates :household_excise_credit_claimed, inclusion: { in: %w[yes no], message: :blank }
+    validate :credit_claimed
+    validate :incarcerated_columns
+
+    # TODO: stop accepting old column
+    def incarcerated_columns
+      if was_incarcerated.present?
```
Comment (embarnard) [inline app/forms/state_file/az_excise_credit_form.rb:18]: check what params are present in these cases

File: app/forms/state_file/az_excise_credit_form.rb [hunk 2]
```diff
@@ -1,13 +1,52 @@
...
+        end
+      end
+    end
+
+    # TODO: replace with:
+    #  validates_presence_of :household_excise_credit_claimed_amt, if: :household_excise_credit_claimed
+    #  validates :household_excise_credit_claimed_amt, numericality: { only_integer: true, greater_than: 0 }, allow_blank: true
+    def credit_claimed
+      if was_incarcerated.blank? && household_excise_credit_claimed == "yes"
+        if household_excise_credit_claimed_amt.blank? || household_excise_credit_claimed_amt <= 0 || !household_excise_credit_claimed_amt.is_a?(Integer)
```
Comment (embarnard) [inline app/forms/state_file/az_excise_credit_form.rb:38]: Add error message for non-integer case

File: app/forms/state_file/az_excise_credit_form.rb [hunk 3]
```diff
@@ -1,13 +1,52 @@
...
+          if spouse_was_incarcerated.blank? || !%w[yes no].include?(spouse_was_incarcerated)
+            errors.add(:spouse_was_incarcerated, I18n.t("errors.messages.blank"))
+          end
+        end
+      end
+    end
+
+    # TODO: replace with:
+    #  validates_presence_of :household_excise_credit_claimed_amt, if: :household_excise_credit_claimed
+    #  validates :household_excise_credit_claimed_amt, numericality: { only_integer: true, greater_than: 0 }, allow_blank: true
```
Comment (embarnard) [inline app/forms/state_file/az_excise_credit_form.rb:35]: also replace with validation for excise credit amount presence if claimed is yes

File: app/forms/state_file/az_excise_credit_form.rb [hunk 4]
```diff
@@ -1,13 +1,52 @@
...
+    def credit_claimed
+      if was_incarcerated.blank? && household_excise_credit_claimed == "yes"
```
Comment (embarnard) [inline app/forms/state_file/az_excise_credit_form.rb:37]: check params here too

File: app/forms/state_file/az_excise_credit_form.rb [hunk 5]
```diff
@@ -1,13 +1,52 @@
 module StateFile
   class AzExciseCreditForm < QuestionsForm
-    set_attributes_for :intake, :was_incarcerated, :ssn_no_employment, :household_excise_credit_claimed
+    set_attributes_for :intake,
```
Comment (embarnard) [inline app/forms/state_file/az_excise_credit_form.rb:3]: could we just refresh the page instead and make them fill it out again?

File: app/lib/efile/az/az140.rb [hunk 6]
```diff
@@ -234,13 +234,14 @@ def calculate_line_53
 
 
       def calculate_line_56
-        if !@direct_file_data.claimed_as_dependent? && @intake.qualified_for_excise_credit? && @intake.ask_whether_incarcerated?
+        if @intake.qualified_for_excise_credit?
```
Comment (jenny-heath) [inline app/lib/efile/az/az140.rb:237]: idea: 
- add method to intake called something like `disqualified_for_excise_credit_federal?`
- call `disqualified_for_excise_credit_federal?` in controller show
- add method to intake called something like `disqualified_for_excise_credit_state?`
- call both methods here?

File: app/models/state_file_az_intake.rb [hunk 7]
```diff
@@ -164,10 +164,16 @@ def disqualifying_eligibility_rules
     }
   end
 
-  def ask_whether_incarcerated?
-    has_valid_ssn = primary.ssn.present? && !primary.has_itin?
-    has_valid_agi = direct_file_data.fed_agi <= (filing_status_mfj? || filing_status_hoh? ? 25_000 : 12_500)
-    has_valid_ssn && has_valid_agi
+  def disqualified_from_excise_credit_df?
```
Comment (embarnard) [inline app/models/state_file_az_intake.rb:167]: niceee these are so much better

File: app/lib/efile/az/az140.rb [hunk 8]
```diff
@@ -238,16 +238,14 @@ def calculate_line_56
...
           number_of_filers = filing_status_mfj? ? 2 : 1
           wrksht_line_2 = number_of_filers - @intake.incarcerated_filer_count
           wrksht_line_4 = (@dependent_count + wrksht_line_2) * 25
 
-          max_credit = @intake.household_excise_credit_claimed_yes? && @intake.household_excise_credit_claimed_amt.is_a?(Integer) ?
-                         100 - @intake.household_excise_credit_claimed_amt :
-                         100
-          return [wrksht_line_4, max_credit].min
+          max_credit = 100
+          max_credit -= @intake.household_excise_credit_claimed_amt if @intake.household_excise_credit_claimed_yes? && @intake.household_excise_credit_claimed_amt.is_a?(Integer)
```
Comment (jenny-heath) [inline app/lib/efile/az/az140.rb:246]: i like

**Pull request merged**